### PR TITLE
executors: add node_exporter dashboards for executor and docker mirror nodes

### PR DESCRIPTION
**Pull request by Strum355**
image

image

### CODE REVIEW

File: monitoring/definitions/executor.go [hunk 1]
```diff
@@ -32,6 +32,21 @@ func Executor() *monitoring.Container {
 				Query: "batches,codeintel",
 				Type:  "custom",
 			},
+			{
+				Label:      "Instance",
+				Name:       "instance",
+				AllValue:   ".*",
+				IncludeAll: true,
+				Query:      "label_values(node_exporter_build_info{job=\"sourcegraph-code-intel-indexer-nodes\"}, instance)",
```
Comment (eseliger) [inline monitoring/definitions/executor.go:40]: should we use `queueContainerName` here from l 14?

(Btw, the comment is wrong and we should remove frontend and worker from the list 👀)

File: monitoring/definitions/shared/node_exporter.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+				},
+				{
+					Name:           "node_cpu_saturation_load5",
+					Description:    "host CPU saturation (5min average)",
+					Query:          "sum(node_load5{job=\""+job+"\",instance=~\"$instance\"}) by (instance) / count(node_cpu_seconds_total{job=\""+job+"\",mode=\"system\",instance=~\"$instance\"}) by (instance) * 100",
+					NoAlert:        true,
+					Interpretation: "banana",
+					Panel:          monitoring.Panel().LegendFormat("{{instance}}").Unit(monitoring.Percentage),
+				},
+			}, */
```
Comment (eseliger) [inline monitoring/definitions/shared/node_exporter.go:52]: Commented out on purpose?

File: monitoring/definitions/shared/node_exporter.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+				/* {
+						Name:           "node_memory_saturation",
+						Description:    "host memory saturation (major page fault rate)",
+						Query:          "sum(rate(node_vmstat_pgmajfault{job=\""+job+"\",instance=~\"$instance\"}[2m])) by (instance)",
+						NoAlert:        true,
+						Interpretation: "banana",
+						Panel:          monitoring.Panel().LegendFormat("{{instance}}"),
+					},
+				},
+				{ */
```
Comment (eseliger) [inline monitoring/definitions/shared/node_exporter.go:77]: Commented out on purpose?

File: monitoring/definitions/shared/node_exporter.go [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+				/* },
+				{
+					{
+						Name:           "node_io_disk_saturation_pressure_some",
+						Description:    "disk IO saturation (some-processes time waiting)",
+						Query:          "rate(node_pressure_io_waiting_seconds_total{job=\""+job+"\",instance=~\"$instance\"}[5m])-rate(node_pressure_io_stalled_seconds_total{job=\""+job+"\",instance=~\"$instance\"}[5m])",
+						NoAlert:        true,
+						Interpretation: "banana",
+						Panel:          monitoring.Panel().LegendFormat("{{instance}}").Unit(monitoring.Seconds),
+					}, */
```
Comment (eseliger) [inline monitoring/definitions/shared/node_exporter.go:129]: Commented out on purpose?

File: monitoring/definitions/shared/node_exporter.go [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+const TitleNodeExporter = "Executor: %s instance metrics"
+
+func NewNodeExporterGroup(containerName, job, jobTitle string) monitoring.Group {
+	return monitoring.Group{
+		Title:  fmt.Sprintf(TitleNodeExporter, jobTitle),
+		Hidden: false,
+		Rows: []monitoring.Row{
+			{
+				{
```
Comment (eseliger) [inline monitoring/definitions/shared/node_exporter.go:17]: So this is total CPU, including guest, right? 
Just a (probably stupid) question: Is it even possible to track CPU by guest?

**Review by eseliger — APPROVED** [review]
Looks all super nice! Very stoked to see what happens inside executors!

File: monitoring/definitions/shared/node_exporter.go (hunk 2 above)
Comment (Strum355) [reply monitoring/definitions/shared/node_exporter.go:52]: Ive a number of panels commented out that I want to keep around for reference. Might move them all collectively to the bottom of the file. What do you think?

Comment (eseliger) [reply monitoring/definitions/shared/node_exporter.go:52]: Works for me. Can you add a comment that this is there on purpose?

File: monitoring/definitions/executor.go (hunk 1 above)
Comment (Strum355) [reply monitoring/definitions/executor.go:40]: Cant use `queueContainerName` here, the current names are as follows:
image

Ive removed frontend and worker from the list though :+1: cc @efritz any reason they were in queueContainerName?

**Review by efritz — APPROVED** [review]

File: monitoring/definitions/executor.go (hunk 1 above)
Comment (Strum355) [reply monitoring/definitions/executor.go:40]: Ok turns out `frontend|sourcegraph-frontend|worker` was in queueContainerName for a reason

Comment (eseliger) [reply monitoring/definitions/executor.go:40]: 👀 :D

File: monitoring/definitions/shared/node_exporter.go (hunk 5 above)
Comment (Strum355) [reply monitoring/definitions/shared/node_exporter.go:17]: Ok this took a while to properly track down: this includes guests, yes https://github.com/munin-monitoring/munin/issues/648#issuecomment-174335344
> Is it even possible to track CPU by guest?
not with node_exporter itself, we'd have to find a different path for this

Comment (eseliger) [reply monitoring/definitions/shared/node_exporter.go:17]: Got it, thanks for clarifying!

**Pull request merged**